theano/gpuarray/magma_cholesky.c

```diff
@@ -0,0 +1,129 @@
+#section kernels
+
+#kernel tril_kernel : size, size, *:
+
+KERNEL void tril_kernel(const ga_size nthreads, const ga_size ncols,
+                        GLOBAL_MEM DTYPE_INPUT_0 *a) {
```

Comment:
You would have to handle the offset on a.

To do this add an extra parameter called something like `a_off` and add this line at the beginning of the kernel code:

```
a = (GLOBAL_MEM DTYPE_INPUT_0 *)(((char *)a) + a_off);
```

You also need to pass the offset for the array when calling the kernel.